src/test/java/com/github/commonsrdf/dummyimpl/LiteralImpl.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.commonsrdf.dummyimpl;
+
+import java.util.Optional;
+
+import com.github.commonsrdf.api.IRI;
+import com.github.commonsrdf.api.Literal;
+
+public class LiteralImpl implements Literal {
+
+	private static final String QUOTE = "\"";
+	private static final IRIImpl RDF_LANG_STRING = new IRIImpl(
+			"http://www.w3.org/1999/02/22-rdf-syntax-ns#langString");
+	private static final IRIImpl XSD_STRING = new IRIImpl(
+			"http://www.w3.org/2001/XMLSchema#string");
+
+	private IRI dataType;
+	private Optional<String> languageTag;
+	private String lexicalForm;
+
+	public LiteralImpl(String literal) {
+		if (literal == null) {
+			throw new NullPointerException("literal can't be null");
+		}
+		this.lexicalForm = literal;
+		this.dataType = XSD_STRING;
+		this.languageTag = Optional.empty();
+	}
+
+	public LiteralImpl(String lexicalForm, IRI dataType) {
+		if (lexicalForm == null) {
+			throw new NullPointerException("lexicalForm can't be null");
+		}
+		this.lexicalForm = lexicalForm;
+		if (dataType == null) {
+			throw new NullPointerException("dataType can't be null");
+		}
+		this.dataType = dataType;
+		this.languageTag = Optional.empty();
+	}
+
+	public LiteralImpl(String literal, String languageTag) {
+		if (literal == null) {
+			throw new NullPointerException("literal can't be null");
+		}
+		this.lexicalForm = literal;
+		this.languageTag = Optional.of(languageTag);
```

Comment:
Could you check the syntax of languageTag against BCP47 using Locale.forLanguageTag?

http://docs.oracle.com/javase/tutorial/i18n/locale/create.html